modules/javafx.graphics/src/main/java/com/sun/glass/ui/win/WinTextRangeProvider.java

```diff
@@ -362,7 +362,7 @@ private String GetText(int maxLength) {
         String text = (String)getAttribute(TEXT);
         if (text == null) return null;
         validateRange(text);
-        int endOffset = maxLength != -1 ? Math.min(end, start + maxLength) : end;
+        int endOffset = maxLength >= 0 ? Math.min(end, Math.max(start + maxLength, maxLength)) : end;
```

Comment:
there is a similar code on LL 101,102,381,485
would we need to apply the same logic there?

Reply:
Done! Hopefully, not too overengeered - https://github.com/openjdk/jfx/pull/1442/files#diff-cf00c7b3f2ccd166df5df3aabcd0940072739356910a2c2107a9442aec6010dfR105

(The JDK code seems to be very compact)

Reply:
I implemented it for L101, felt a bit bad and at https://github.com/openjdk/jfx/pull/1442/files/e7502d674c579667545b97ab944e82f2b093a470#diff-cf00c7b3f2ccd166df5df3aabcd0940072739356910a2c2107a9442aec6010df, Kevin also noted it could not be right there. Thus, I reverted to the original.

For the others, I used the new utility method `getValidStringIndex`